src/Syntax/CST/Terms.hs

```diff
@@ -1,42 +1,208 @@
 
+--------------------------------------------
+
+test1 :: [Pattern]
+test1 =
+  [ PatXtor defaultLoc (MkXtorName (pack "Leaf")) [PatVar defaultLoc (MkFreeVarName (pack "x"))],
+    PatXtor
+      defaultLoc
+      (MkXtorName (pack "Branch"))
+      [ PatXtor defaultLoc (MkXtorName (pack "Leaf")) [PatVar defaultLoc (MkFreeVarName (pack "y"))],
+        PatVar defaultLoc (MkFreeVarName (pack "t2"))
+      ],
+    PatVar defaultLoc (MkFreeVarName (pack "x")),
+    PatStar defaultLoc,
+    PatWildcard defaultLoc,
+    PatXtor
+      defaultLoc
+      (MkXtorName (pack "Branch"))
+      [ PatXtor defaultLoc (MkXtorName (pack "Leaf")) [PatVar defaultLoc (MkFreeVarName (pack "y"))],
+        PatXtor defaultLoc (MkXtorName (pack "Leaf")) [PatVar defaultLoc (MkFreeVarName (pack "z"))]
+      ],
+    PatXtor
+      defaultLoc
+      (MkXtorName (pack "Branch"))
+      [ PatVar defaultLoc (MkFreeVarName (pack "t1")),
+        PatVar defaultLoc (MkFreeVarName (pack "t2"))
+      ],
+    PatVar defaultLoc (MkFreeVarName (pack "t"))
+  ]
+
+test2 :: [Pattern]
+test2 =
+  [ PatVar defaultLoc (MkFreeVarName (pack "m")),
+    PatStar defaultLoc,
+    PatXtor defaultLoc (MkXtorName (pack "Nothing")) [],
+    PatXtor defaultLoc (MkXtorName (pack "Maybe")) [PatVar defaultLoc (MkFreeVarName (pack "x"))],
+    PatWildcard defaultLoc
+  ]
+
+test3 :: [Pattern]
+test3 = []
+
+test4 :: [Pattern]
+test4 = [PatStar defaultLoc]
+
+test5 :: [Pattern]
+test5 =
+  [ PatXtor
+      defaultLoc
+      (MkXtorName (pack "Node"))
+      [ PatXtor defaultLoc (MkXtorName (pack "Subtree")) [PatVar defaultLoc (MkFreeVarName (pack "y"))],
+        PatXtor defaultLoc (MkXtorName (pack "Subtree")) [PatVar defaultLoc (MkFreeVarName (pack "x"))],
+        PatVar defaultLoc (MkFreeVarName (pack "z"))
+      ]
+  ]
+
+test6 :: [Pattern]
+test6 =
+  [ PatVar defaultLoc (MkFreeVarName (pack "x")),
+    PatVar defaultLoc (MkFreeVarName (pack "z")),
+    PatVar defaultLoc (MkFreeVarName (pack "x"))
+  ]
+
+test7 :: [Pattern]
+test7 = [PatXtor defaultLoc (MkXtorName (pack "Cons")) 
+          [PatVar defaultLoc (MkFreeVarName (pack "x")),
+          PatXtor defaultLoc (MkXtorName (pack "Cons")) 
+            [PatVar defaultLoc (MkFreeVarName (pack "y")),
+            PatXtor defaultLoc (MkXtorName (pack "Cons")) [PatVar defaultLoc (MkFreeVarName (pack "z")), PatVar defaultLoc (MkFreeVarName (pack "zs"))]]],
+         PatXtor defaultLoc (MkXtorName (pack "Cons")) 
+          [PatVar defaultLoc (MkFreeVarName (pack "x")),
+          PatXtor defaultLoc (MkXtorName (pack "Cons")) 
+            [PatVar defaultLoc (MkFreeVarName (pack "y")),
+            PatXtor defaultLoc (MkXtorName (pack "Cons")) 
+              [PatVar defaultLoc (MkFreeVarName (pack "z")),
+               PatXtor defaultLoc (MkXtorName (pack "Cons")) [PatVar defaultLoc (MkFreeVarName (pack "m")), PatVar defaultLoc (MkFreeVarName (pack "ms"))]]]]]
+
+
+--An Overlap Message is a String
+type OverlapMsg = String
```

Comment:
In Haskell wollen wir eigentlich nie den Typ `String` verwenden, da der Typ sehr ineffizient ist. Ein String ist eine linked list von Charactern. Der effiziente Typ für Strings ist `Text` aus `Data.Text`, für den du auch die gleichen Literale verwenden kannst.

Reply:
Okay, das habe ich jetzt geändert